src/core/scroll/scroll.service.js

```diff
@@ -0,0 +1,158 @@
+import { EventListener } from '../infrastructure/event.listener';
+import { EventManager } from '../infrastructure/event.manager';
+import { Disposable } from '../infrastructure/disposable';
+import { AppError } from '../infrastructure/error';
+import { jobLine } from '../services/job.line';
+
+const OFFSET = 50;
+const VELOCITY = 5;
+
+export class ScrollService extends Disposable {
+    constructor(model, table) {
```

Comment:
disposable service is not good practice, move all disposables outside of the service and body ctrl